libcontainer/cgroups/fs2/hugetlb.go

```diff
@@ -0,0 +1,46 @@
+// +build linux
+
+package fs2
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/opencontainers/runc/libcontainer/cgroups"
+	"github.com/opencontainers/runc/libcontainer/cgroups/fscommon"
+	"github.com/opencontainers/runc/libcontainer/configs"
+)
+
+func setHugeTlb(dirPath string, cgroup *configs.Cgroup) error {
+	for _, hugetlb := range cgroup.Resources.HugetlbLimit {
+		if err := fscommon.WriteFile(dirPath, strings.Join([]string{"hugetlb", hugetlb.Pagesize, "max"}, "."), strconv.FormatUint(hugetlb.Limit, 10)); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func statHugeTlb(dirPath string, stats *cgroups.Stats, cgroup *configs.Cgroup) error {
```

Comment:
@odinuge Sorry, didn't get it, can you explain what do you mean by that?

Reply:
Ahh, sorry!

The `MaxUsage` field for hugepages is not present in cgroupv2. It is still a part of the struct `cgroups.HugetlbStats`, so i thought it would be nice to add a comment that it is only populated when using cgroupv2.

https://github.com/opencontainers/runc/blob/2c632d1a2de0192c3f18a2542ccb6f30a8719b1f/libcontainer/cgroups/stats.go#L89-L90


Maybe add a comment with something like this:

```go
 // maximum usage ever recorded. Only for cgroup v1
 MaxUsage uint64 `json:"max_usage,omitempty"` 
```